server/services/v1/query_runner.go

```diff
@@ -591,17 +580,31 @@ func (runner *SearchQueryRunner) Run(ctx context.Context, tx transaction.Tx, ten
 					UpdatedAt: row.Data.UpdatedToProtoTS(),
 				},
 			})
-			totalInPage++
 
 			if len(resp.Hits) == pageSize {
 				break
 			}
 		}
-		if len(resp.Hits) == 0 {
+		if rowReader.err != nil {
+			return nil, ctx, rowReader.err
+		}
+		resp.Facets = rowReader.getFacets()
+		ps := int64(pageSize)
+		totalPages := int32(rowReader.getTotalFound()/ps) + 1
+		resp.Meta = &api.SearchMetadata{
+			Found:      rowReader.getTotalFound(),
+			TotalPages: totalPages,
+			Page: &api.Page{
+				Current: pageNo,
+				Size:    int32(searchQ.PageSize),
+			},
+		}
+
+		// break the stream after sending at least 1 response
+		if len(resp.Hits) == 0 && pageNo != int32(defaultPageNo) {
```

Comment:
Shouldn't the condition here be OR and also what if the user passed the pageNo as `1` then also we need to stop?